crates/isograph_schema/src/validate_schema.rs

```diff
@@ -440,6 +440,33 @@ fn validate_and_transform_client_fields(
     }))
 }
 
+fn validate_all_variables_are_used(
+    variable_definitions: Vec<WithSpan<UnvalidatedVariableDefinition>>,
+    used_variables: Vec<VariableName>,
```

Comment:
love that we're consuming it here, this forces us to collect all used variables before we validate